app/src/common/chromium/com/igalia/wolvic/browser/api/impl/PromptDelegateImpl.java

```diff
@@ -490,4 +493,66 @@ public void hide(boolean sendsCancelMessage) {
             mChoicePrompt.dismiss();
         }
     }
+
+    public class ColorChooserFactory implements ColorChooserManager.Factory {
+        public ColorChooserManager.Bridge create(int initialColor, ColorChooserManager.Listener listener) {
+            return new ColorPromptBridge(initialColor, listener);
+        }
+    }
+
+    public class ColorPromptBridge implements ColorChooserManager.Bridge {
+
+        private final ColorPrompt mColorPrompt;
+
+        public ColorPromptBridge(int initialColor, ColorChooserManager.Listener listener) {
+            mColorPrompt = new ColorPrompt(initialColor, listener);
+        }
+
+        @Override
+        public void show() {
+            try {
+                final WSession.PromptDelegate delegate = mSession.getPromptDelegate();
+                delegate.onColorPrompt(mSession, mColorPrompt);
+            } catch (WindowManager.BadTokenException e) {
+                mColorPrompt.markComplete();
+            }
+        }
+
+        @Override
+        public void close() {
+            if (!mColorPrompt.isComplete())
+                mColorPrompt.dismiss();
+        }
+    }
+
+    public static class ColorPrompt extends BasePromptImpl implements WSession.PromptDelegate.ColorPrompt {
+        private final String mDefaultColor;
+        private final ColorChooserManager.Listener mListener;
+
+        public ColorPrompt(int defaultColor, ColorChooserManager.Listener listener) {
+            mDefaultColor = "#" + Integer.toHexString(defaultColor);
+            mListener = listener;
+        }
+
+        @Nullable
+        @Override
+        public String defaultValue() { return mDefaultColor; }
+
+        @UiThread
+        @NonNull
+        @Override
+        public WSession.PromptDelegate.PromptResponse confirm(@NonNull final String color) {
+            markComplete();
+            mListener.onColorChanged(Color.parseColor(color));
+            return new PromptResponseImpl();
+        }
+
+        @NonNull
+        @Override
+        public WSession.PromptDelegate.PromptResponse dismiss() {
+            if (!isComplete())
```

Comment:
Not sure, but probably we need markComplete() here?

Reply:
No problems were found when testing, but added this code by referencing other code. thanks!